pkg/providers/sqs/sqs.go

```diff
@@ -0,0 +1,98 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sqs
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/sqs"
+	"github.com/aws/aws-sdk-go/service/sqs/sqsiface"
+
+	"github.com/aws/karpenter/pkg/operator/options"
+)
+
+type Provider struct {
+	client sqsiface.SQSAPI
+
+	name string
+	url  string
+}
+
+func NewProvider(ctx context.Context, client sqsiface.SQSAPI) (*Provider, error) {
+	ret, err := client.GetQueueUrlWithContext(ctx, &sqs.GetQueueUrlInput{
+		QueueName: aws.String(options.FromContext(ctx).InterruptionQueue),
```

Comment:
This constructor's method signature feels a bit magical. Any reason to not pass the queue name in as an argument?

Reply:
Agreed, good callout. I was going back and forth on it, but I think your suggestion is best